internal/services/metadata/store.go

```diff
@@ -0,0 +1,54 @@
+package metadata
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"castai-agent/internal/config"
+)
+
+type Store interface {
+	// StoreMetadataSecret stores relevant agent runtime metadata in a secret.
+	StoreMetadataSecret(ctx context.Context, metadata *Metadata) error
+}
+
+var _ Store = (*StoreImpl)(nil)
+
+type StoreImpl struct {
+	clientset kubernetes.Interface
+	cfg       config.Config
+}
+
+func New(clientset kubernetes.Interface, cfg config.Config) *StoreImpl {
+	return &StoreImpl{
+		clientset: clientset,
+		cfg:       cfg,
+	}
+}
+
+func (s *StoreImpl) StoreMetadataSecret(ctx context.Context, metadata *Metadata) error {
+	if !s.cfg.MetadataStore.Enabled {
+		return nil
+	}
+
+	secret := &corev1.Secret{
```

Comment:
Do we plan storing any sensitive data here? If not, maybe it would be better to use configMap instead: it is easier to inspect the values with `kubectl get cm -o yaml` since you don't need to decode the b64.

Reply:
Currently there are no plans to store sensitive data and I agree that config map seems like a better store in that case. Cluster id is intended to be used by cluster controller, which currently only supports loading it from a secret and not from a config map that's why I'm storing it in a secret instead